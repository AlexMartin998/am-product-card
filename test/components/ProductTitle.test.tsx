import * as React from 'react';
import { create } from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { testProduct, testProduct2 } from '../data/product';

describe('it', () => {
  it('should return a correctly rendered component', () => {
    const wrapper = create(
      <ProductTitle title="Custom Product" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should return a component with product name', () => {
    const wrapper = create(
      <ProductCard product={testProduct}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
