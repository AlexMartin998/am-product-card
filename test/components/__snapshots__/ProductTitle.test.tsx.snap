// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should return a component with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug - card Without Img
  </span>
</div>
`;

exports[`it should return a correctly rendered component 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;
